chainstate/src/test_framework/framework.rs

```diff
@@ -146,14 +140,104 @@ impl TestFramework {
     }
 }
 
+fn create_utxo_data(
+    outsrc: OutPointSourceId,
+    index: usize,
+    output: &TxOutput,
+    rng: &mut impl Rng,
+) -> Option<(TxInput, TxOutput)> {
+    Some((
+        TxInput::new(outsrc, index as u32, empty_witness(rng)),
+        match output.value() {
+            OutputValue::Coin(output_value) => {
+                let spent_value = Amount::from_atoms(rng.gen_range(0..output_value.into_atoms()));
+                let new_value = (*output_value - spent_value).unwrap();
+                utils::ensure!(new_value >= Amount::from_atoms(1));
+                TxOutput::new(
+                    OutputValue::Coin(new_value),
+                    OutputPurpose::Transfer(anyonecanspend_address()),
+                )
+            }
+        },
+    ))
+}
+
+pub fn empty_witness(rng: &mut impl Rng) -> InputWitness {
+    use crypto::random::SliceRandom;
+    let mut msg: Vec<u8> = (1..100).collect();
+    msg.shuffle(rng);
+    InputWitness::NoSignature(Some(msg))
+}
+
+pub fn anyonecanspend_address() -> Destination {
+    Destination::AnyoneCanSpend
+}
+
+pub(crate) fn create_new_outputs(
+    srcid: OutPointSourceId,
+    outs: &[TxOutput],
+    rng: &mut impl Rng,
+) -> Vec<(TxInput, TxOutput)> {
+    outs.iter()
+        .enumerate()
+        .filter_map(move |(index, output)| create_utxo_data(srcid.clone(), index, output, rng))
+        .collect()
+}
 impl Default for TestFramework {
     fn default() -> Self {
         Self::builder().build()
     }
 }
+use common::chain::OutPointSourceId;
+// TODO: Replace by a proper UTXO set abstraction
+// (https://github.com/mintlayer/mintlayer-core/issues/312).
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct TestBlockInfo {
+    pub(crate) txns: Vec<(OutPointSourceId, Vec<TxOutput>)>,
+    pub(crate) id: Id<GenBlock>,
+}
+
+impl TestBlockInfo {
+    pub(crate) fn from_block(blk: &Block) -> Self {
+        let txns = blk
+            .transactions()
+            .iter()
+            .map(|tx| {
+                (
+                    OutPointSourceId::Transaction(tx.get_id()),
+                    tx.outputs().clone(),
+                )
+            })
+            .collect();
+        let id = blk.get_id().into();
+        Self { txns, id }
+    }
+
+    pub(crate) fn from_genesis(genesis: &Genesis) -> Self {
+        let id: Id<GenBlock> = genesis.get_id().into();
+        let outsrc = OutPointSourceId::BlockReward(id);
+        let txns = vec![(outsrc, genesis.utxos().to_vec())];
+        Self { txns, id }
+    }
+
+    pub(crate) fn from_id(cs: &TestChainstate, id: Id<GenBlock>) -> Self {
+        match id.classify(&cs.chain_config) {
+            GenBlockId::Genesis(_) => Self::from_genesis(cs.chain_config.genesis_block()),
+            GenBlockId::Block(id) => {
+                let block = cs.chainstate_storage.get_block(id).unwrap().unwrap();
```

Comment:
Clippy is configured to allow `unwrap` in the testing mode but not in production code. Since this is now not guarded by `#[cfg(test)]`, it starts complaining. You may want to add `#![allow(clippy::unwrap_used)]` to the top of the `test_framework` module.